web3/contract.py

```diff
@@ -315,6 +362,58 @@ def factory(cls, web3, class_name=None, **kwargs):
         contract.fallback = Contract.get_fallback_function(contract.abi, contract.web3)
 
         return contract
+    
+    @classmethod
+    def link_bytecode(cls, attr_dict: Dict[str, str]) -> Type["LinkableContract"]:
```

Comment:
Typing wise, I believe this is the better way to do this.

```python
TContract = TypeVar("TContract", bound="Contract")

class Contract:
    @classmethod
    def link_bytecode(cls: Type[TContract], ...) -> Type[TContract]:
        ...
```

I *think* that this results in it working properly for subclasses.

Also, `LinkableContract` is no longer a thing right?